emop-maven-plugin/src/main/java/edu/cornell/VmsMojo.java

```diff
@@ -0,0 +1,277 @@
+    }
+
+    /**
+     * Updates the lineChanges and renames based on found differences.
+     *
+     * @param diffs List of differences between two versions of the same program
+     * @throws MojoExecutionException if error is encountered during runtime
+     */
+    private void findLineChangesAndRenames(List<DiffEntry> diffs) throws MojoExecutionException {
+        try {
+            for (DiffEntry diff : diffs) {
+                // If the old path is /dev/null, then the file has been created between commits
+                if (diff.getOldPath().equals("/dev/null")) {
+                    newClasses.add(diff.getNewPath());
+                } else if (!diff.getNewPath().equals("/dev/null")) { // Ignore if a deleted class
+                    // Gets renamed classes
+                    if (!diff.getOldPath().equals(diff.getNewPath())) {
+                        renames.put(diff.getNewPath(), diff.getOldPath());
+                    }
+
+                    // For each file, find the replacements, additions, and deletions at each line
+                    for (Edit edit : diffFormatter.toFileHeader(diff).toEditList()) {
+                        Map<Integer, Integer> lineChange = new HashMap<>();
+                        if (lineChanges.containsKey(diff.getOldPath())) {
+                            lineChange = lineChanges.get(diff.getOldPath());
+                        }
+                        lineChange.put(edit.getBeginA() + 1, edit.getLengthB() - edit.getLengthA());
+                        lineChanges.put(diff.getOldPath(), lineChange);
+                    }
+                }
+            }
+        } catch (IOException exception) {
+            throw new MojoExecutionException("Encountered an error when comparing different files");
+        }
+    }
+
+    /**
+     * Scrubs newViolations of violations believed to be duplicates from violation-counts-old.
+     */
+    private void removeDuplicateViolations() {
+        Set<Violation> violationsToRemove = new HashSet<>();
+        for (Violation newViolation : newViolations) {
+            Set<Violation> relevantOldViolations = oldViolations.stream()
+                    .filter(oldViolation -> oldViolation.getSpecification().equals(newViolation.getSpecification()))
+                    .filter(oldViolation -> oldViolation.getClassInfo().equals(newViolation.getClassInfo())
+                                         || isRenamed(oldViolation.getClassInfo(), newViolation.getClassInfo()))
+                    .filter(oldViolation -> hasSameLineNumber(oldViolation.getClassInfo(), oldViolation.getLineNum(),
+                            newViolation.getLineNum()))
+                    .collect(Collectors.toSet());
+
+            if (!relevantOldViolations.isEmpty()) {
+                violationsToRemove.add(newViolation);
+            }
+        }
+        for (Violation violationToRemove : violationsToRemove) {
+            newViolations.remove(violationToRemove);
+        }
```

Comment:
Replace lines 170 to 172 with something `newViolations.removeAll(violationsToRemove);`